designs/2018-processors-improvements/README.md

```diff
@@ -0,0 +1,202 @@
+        markdown: {
+            preprocess() {},
+            postprocess() {}
+        }
+    }
+};
+```
+
+This plugin now exports a named `markdown` processor that can be referenced directly from configuration. Existing plugins would need to be changed, but can be done in a backwards-compatible way (see Backwards Compatibility section below).
+
+### Add a `processors` Configuration Option
+
+In order to use a named processor, end-users would specify a `processors` option in their configuration. The option must be an array and must contain at least one string identifier. The string identifier must correspond to a named processor from a plugin and have the form *pluginName/processorName*, which would reference the `pluginName` processor in the `eslint-plugin-pluginName` plugin. For example: 
+
+```js
+module.exports = {
+    plugins: ["markdown"],
+    parserOptions: {
+        ecmaVersion: 2018
+    },
+    processors: ["markdown/markdown"]
+};
+```
+
+This configuration file will apply the named processor `markdown` from the `eslint-plugin-markdown` plugin to all files.
+
+### Apply Processors Using `overrides`
+
+Of course, in most cases the end-user will not want to apply a processor to all files. In that case, the `overrides` configuration option may be used to narrowly define to which files the processors should be applied. For example:
+
+```js
+module.exports = {
+    plugins: ["markdown"],
+    parserOptions: {
+        ecmaVersion: 2018
+    },
+    overrides: [
+        {
+            files: ["*.md"],
+            processors: ["markdown/markdown"]
+        }
+    ]
+};
+```
+
+In this configuration, the named processor `markdown` from the `eslint-plugin-markdown` plugin is applied only to files with the extension `*.md`. This mimics the behavior of the extension-named processor that `eslint-plugin-markdown` currently exports.
+
+### Apply Multiple Processors
+
+In special cases, end-users may want to apply multiple processors to files. This can be accomplished by listing multiple named processors using the `processors` configuration option. For example:
+
+```js
+module.exports = {
+    plugins: ["markdown", "vue"],
+    parserOptions: {
+        ecmaVersion: 2018
+    },
+    overrides: [
+        {
+            files: ["*.md"],
+            processors: ["markdown/markdown", "vue/vue"]
+        }
+    ]
+};
+```
+
+This configuration applies two named processors to all files matching `*.md`. First, the `markdown` named processor from `eslint-plugin-markdown` is applied. The result of that preprocessing is then passed to the `vue` named processor from `eslint-plugin-vue`. The result of that preprocessing is then passed to ESLint for linting. When ESLint has completed linting, the reverse sequence occurs: the linting results are first passed to the `vue` named processor from `eslint-plugin-vue`, and once that postprocessing is complete, the result is then passed to the `markdown` named processor from  `eslint-plugin-markdown`.
```

Comment:
I think it might be worth noting that this requires the `vue` processor to be able to handle non-Vue/non-JS code gracefully. (For example, if the Markdown processor passed along all code blocks rather than just JS, then the Vue plugin needs to gracefully handle that.)

Reply:
Yes, I'll update that.